exporters/otlp/common/src/main/java/io/opentelemetry/exporter/otlp/internal/MarshalerUtil.java

```diff
@@ -177,18 +177,26 @@ public static int sizeInt32(ProtoFieldInfo field, int message) {
 
   /** Returns the size of a double field. */
   public static int sizeDouble(ProtoFieldInfo field, double value) {
-    if (value == 0D) {
+    return sizeDouble(field, value, /* force= */ false);
+  }
+  /** Returns the size of a double field. */
+  public static int sizeDouble(ProtoFieldInfo field, double value, boolean force) {
+    if (!force && value == 0D) {
       return 0;
     }
     return field.getTagSize() + CodedOutputStream.computeDoubleSizeNoTag(value);
   }
 
   /** Returns the size of a fixed64 field. */
-  public static int sizeFixed64(ProtoFieldInfo field, long message) {
-    if (message == 0L) {
+  public static int sizeFixed64(ProtoFieldInfo field, long value) {
+    return sizeFixed64(field, value, /* force= */ false);
+  }
+  /** Returns the size of a fixed64 field. */
+  public static int sizeFixed64(ProtoFieldInfo field, long value, boolean force) {
```

Comment:
We don't need a force flag, the `== 0L` check can remain in the other method and this one can have a name like `sizedFixed64ForOneof` or `sizedRequiredFixed64` (interesting how the serialization is required for an optional field :P)

Reply:
It's not an optional field, it's a oneof.  You actually handle this for [AnyValue](https://github.com/open-telemetry/opentelemetry-java/blob/main/exporters/otlp/common/src/main/java/io/opentelemetry/exporter/otlp/internal/KeyValueMarshaler.java#L143), just missed that here.  Didn't think it was worth making a new Marshaller instance for it.

Reply:
The name can be whatever but I'd avoid this boolean pattern. If we add any optional field, it would call the same method (optional == oneof), hence referring to optional fields.

Reply:
Ah, right.   Basically this is:   proto2 *or* proto3Optional *or* proto3 oneof-primtive.

primitive oneof = proto3optional = proto2 in the case of field-presence being mandatory.

In this case it's just the fact it's a oneof.   I can either call the boolean "fieldPresenceRequired" or somehow denote it that way?

Reply:
I'd prefer separate methods with different names to a boolean, it keeps the logic simpler. An amazing name doesn't come to mind but some form of these suggestions should be ok.

Reply:
I realized an alternative to figure out naming would be to just stick to our current pattern of not calling a helper for oneofs, for now.

https://github.com/open-telemetry/opentelemetry-java/blob/main/exporters/otlp/common/src/main/java/io/opentelemetry/exporter/otlp/internal/KeyValueMarshaler.java#L128

FYI, on the back burner has been changing the codegen to allow something like `Span.NAME.size(nameUtf8)` instead of using util methods, where the method would implement oneof semantics directly where appropriate.

Reply:
That'd be really nice.  I just used "Optional" for now and we can understand that oneof == optional in proto3 encoding.